File: Binary Search/BinarySearch.py
from typing import List

def binary_search(L: List[int], x: int) -> int:
    low, high = 0, len(L) - 1
    
    # Binary search loop
    while low <= high:
        mid = (low + high) // 2  # Integer division
        
        if L[mid] == x:
            return mid  # Found x at position mid
        elif L[mid] < x:
            low = mid + 1  # Search in the right half
        else:
            high = mid - 1  # Search in the left half
    
    return -1  # x not found in L


if __name__ == "__main__":
    L = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

    for x in range(1, len(L)):
        position = binary_search(L, x)
        
        if position == -1:
            print(f"{x} not found in the list.")
        else:
            print(f"{x} found at position {position}.")
    
    print("Program terminated")

File: Binary Search/BinarySearch.java
import java.util.List;

public class BinarySearch {

    public static int binarySearch(List<Integer> L, int x) {
        int low = 0, high = L.size() - 1;

        while (low <= high) {
            int mid = (low + high) / 2;  // Integer division

            if (L.get(mid) == x) {
                return mid;  // Found x at position mid
            } else if (L.get(mid) < x) {
                low = mid + 1;  // Search in the right half
            } else {
                high = mid - 1;  // Search in the left half
            }
        }
        return -1;  // x not found in L
    }

    public static void main(String[] args) {
        List<Integer> L = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);

        for (int x = 1; x <= 10; x++) {
            int position = binarySearch(L, x);
            
            if (position == -1) {
                System.out.println(x + " not found in the list.");
            } else {
                System.out.println(x + " found at position " + position + ".");
            }
        }
        System.out.println("Program terminated.");
    }
}
